src/runtime_src/core/edge/user/aie/AIEResources.cpp

```diff
@@ -0,0 +1,403 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (C) 2020, Xilinx Inc - All rights reserved
+ * ZNYQ XRT Library layered on top of ZYNQ zocl kernel driver
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "AIEResources.h"
+#include <iostream>
+
+#include "core/common/error.h"
+#include "core/common/message.h"
+
+namespace zynqaie
+{
+namespace Resources
+{
+    /// performanceCounters, traceEvents, eventBroadcasts are short (int16) arrays where individual element stores the event handle (short, int16) that holds the resources.
+    /// The default element value is -1 (invalid_handle) meaning the resources is available.
+    /// NOTE: to reduce memory footprint for AIEResources, short (int16) data type is used. In adf.h, handle is typedef as int.
+    const short invalid_handle = -1;
+    const short ecc_scrubbing_reserve = -2;
```

Comment:
namespace indentation should be avoided.
https://llvm.org/docs/CodingStandards.html#namespace-indentation